src/Reveal.Sdk.Dom.Tests/Visualizations/VisualizationFixture.cs

```diff
@@ -0,0 +1,514 @@
+
+            var document = new RdashDocument();
+            document.DataSources.Add(dataSourceResourceItem);
+            document.Visualizations.Add(visualization);
+
+            RdashDocumentValidator.Validate(document);
+
+            // Act
+            var dataSourceItemResult = visualization.GetDataSourceItem();
+
+            // Assert
+            Assert.Equal(dataSourceItem, dataSourceItemResult);
+            Assert.Equal((visualization.DataDefinition as TabularDataDefinition).Fields.Count, dataSourceItemResult.Fields.Count);
+            Assert.Equal(dataSourceItemResult.DataSource.Id, dataSource.Id);
+            Assert.Equal(dataSourceItemResult.ResourceItemDataSource.Id, dataSourceResourceItem.Id);
+        }
+
+        [Fact]
+        public void UpdateDataSourceItem_UpdateNewDataSourceItem_WithTabularDSItem()
+        {
+            // Arrange
+            var dataSource = new DataSource()
+            {
+                Id = "DataSourceId",
+            };
+
+            var dataSourceItem = new DataSourceItem
+            {
+                DataSource = dataSource,
+                HasTabularData = true,
+                Fields = new List<IField>
+                {
+                    new TextField("A"),
+                    new TextField("B"),
+                }
+            };
+
+            var mock = new Mock<Visualization>("testTitle", dataSourceItem) { CallBase = true };
+            var visualization = mock.Object;
+
+            var newDataSource = new DataSource();
+            var newDataSourceItem = new DataSourceItem
+            {
+                DataSource = newDataSource,
+                HasTabularData = true,
+                Fields = new List<IField>
+                {
+                    new TextField("A1"),
+                    new TextField("B2"),
+                    new TextField("C3"),
+                    new TextField("D4"),
+                }
+            };
+
+            // Mock join tables
+            var leftJoinFieldName = "left";
+            var rightJoinFieldName = "right";
+            var aliasName = "Alias";
+            var joinConditions = new List<JoinCondition> { new JoinCondition(leftJoinFieldName, rightJoinFieldName) };
+            var dataSourceItemToJoin = new DataSourceItem();
+
+            newDataSourceItem.Join(aliasName, leftJoinFieldName, rightJoinFieldName, dataSourceItemToJoin);
+
+            // Act
+            visualization.UpdateDataSourceItem(newDataSourceItem);
+
+            // Assert
+            Assert.Equal(newDataSourceItem, visualization.DataDefinition.DataSourceItem);
+            Assert.Equal(newDataSourceItem.Fields.Count, (visualization.DataDefinition as TabularDataDefinition).Fields.Count);
+            Assert.Equal(newDataSourceItem.JoinTables.Count, (visualization.DataDefinition as TabularDataDefinition).JoinTables.Count);
+        }
+
+        [Fact]
+        public void UpdateDataSourceItem_UpdateNewDataSourceItem_WithNonTabularDsItem()
```

Comment:
We should have another test to test for the case `visualization.UpdateDataSourceItem(null);` The expected behaviour is that the `visualization.DataDefinition.DataSourceItem` will not be changed